scripts/install

```diff
@@ -32,162 +32,137 @@ app=$YNH_APP_INSTANCE_NAME
 #=================================================
 # CHECK IF THE APP CAN BE INSTALLED WITH THESE ARGS
 #=================================================
-ynh_print_info "Validating installation parameters..."
+ynh_script_progression --message="Validating installation parameters..."
 
 final_path=/var/www/$app
-test ! -e "$final_path" || ynh_die "This path already contains a folder"
-
-# Normalize the url path syntax
-path_url=$(ynh_normalize_url_path $path_url)
+test ! -e "$final_path" || ynh_die --message="This path already contains a folder"
 
 # Register (book) web path
-ynh_webpath_register $app $domain $path_url
+ynh_webpath_register --app=$app --domain=$domain --path_url=$path_url
 
 #=================================================
 # STORE SETTINGS FROM MANIFEST
 #=================================================
-ynh_print_info "Storing installation settings..."
+ynh_script_progression --message="Storing installation settings..."
 
-ynh_app_setting_set $app domain $domain
-ynh_app_setting_set $app path $path_url
+ynh_app_setting_set --app=$app --key=domain --value=$domain
+ynh_app_setting_set --app=$app --key=path --value=$path_url
 
 #=================================================
 # STANDARD MODIFICATIONS
 #=================================================
 # FIND AND OPEN A PORT
 #=================================================
-ynh_print_info "Configuring firewall..."
+ynh_script_progression --message="Finding an available port..."
 
-# Find a free port
-port=$(ynh_find_port 4242)
-# Open this port
-#ynh_exec_warn_less yunohost firewall allow --no-upnp TCP $port
-ynh_app_setting_set $app port $port
+# Find an available port
+port=$(ynh_find_port --port=4242)
+ynh_app_setting_set --app=$app --key=port --value=$port
 
 #=================================================
 # INSTALL DEPENDENCIES
 #=================================================
-ynh_print_info "Installing dependencies..."
+ynh_script_progression --message="Installing dependencies..."
 
-ynh_install_app_dependencies python-virtualenv sqlite3 libssl-dev zlib1g-dev libncurses5-dev libncursesw5-dev libreadline-dev libsqlite3-dev
+ynh_install_app_dependencies $pkg_dependencies
 
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
-ynh_print_info "Setting up source files..."
+ynh_script_progression --message="Setting up source files..."
 
-ynh_app_setting_set $app final_path $final_path
+ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 # Download, check integrity, uncompress and patch the source from app.src
-git clone https://github.com/YunoHost/yunorunner "$final_path"
+ynh_setup_source --dest_dir="$final_path"
```

Comment:
I did add that git clone not so long ago though, because it's super boring when we commit on yunorunner to have to update the damn app.src with the checksum and launch a full app upgrade just to propagate a bunch of commits in production :/ But clearly it's not orthodox ... I don't know how to "properly" handle it ...

Reply:
well having the app.src let us be sure of the version installed, that why I switched back to app.src.

A proposition to go back to git but keeping control of the version : #5

In case of upgrade of yunorunner:
manifest.json : `version `has to be updated
README.md : `Shipped version:` has to be updated
_common.sh : `release=`  has to be updated
check_process : an upgrade test from current version has to be added